docs/source-pytorch/advanced/warnings.rst

```diff
@@ -0,0 +1,52 @@
+########
+Warnings
+########
+
+Lightning warns users of possible misconfiguration, performance implications or potential mistakes through the :class:`~lightning.pytorch.utilities.warnings.PossibleUserWarning` category.
+Sometimes these warnings can be false positives, and you may want to suppress them to avoid cluttering the logs.
+
+
+.. warning::
+
+    Suppressing warnings is not recommended in general, because they may raise important issues that you should address.
+    Only suppress warnings if they are false.
+
+
+-----
+
+
+*********************************
+Suppress a single warning message
+*********************************
+
+Suppressing an individual warning message can be done through the :mod:`warnings` module:
+
+.. code-block:: python
+
+    import warnings
+
+    warnings.filterwarnings("ignore", ".*Consider increasing the value of the `num_workers` argument*")
+
+
+-----
+
+
+*********************************************
+Suppress all instances of PossibleUserWarning
+*********************************************
+
+Suppressing all warnings of the :class:`~lightning.pytorch.utilities.warnings.PossibleUserWarning` category can be done programmatically
+
+.. code-block:: python
+
+    import lightning as L
+
+    # ignore all warnings that could be false positives
+    L.disable_possible_user_warnings()
```

Comment:
Is it a good idea to put this directly under L if it wont work for app/data/store... warnings?

Reply:
I don't see it being a big issue. The idea could be applied to all of them if we had PossibleUserWarning warnings in these submodules. But if we think it's misleading, I can restrict it to just pytorch and fabric.

Reply:
We couldn't reuse the category unless we put it in lightning-utilities and all packages install that. I would personally restrict it